core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -944,6 +944,14 @@ public XContentBuilder percentageField(String rawFieldName, String readableField
         return this;
     }
 
+    public XContentBuilder nanosAsMillis(String rawFieldName, String readableFieldName, long nanos) throws IOException {
```

Comment:
can't you use the existing `timeValueField` that takes the TimeUnit too as an argument? Then the conversion to readable should come for free, and the raw field gets printed out as-is.